R/setops.R

```diff
@@ -103,8 +103,8 @@ fsetequal = function(x, y, all=TRUE) {
 # all.equal ----
 
 all.equal.data.table = function(target, current, trim.levels=TRUE, check.attributes=TRUE, ignore.col.order=FALSE, ignore.row.order=FALSE, tolerance=sqrt(.Machine$double.eps), ...) {
-  stopifnot(is.logical(trim.levels), is.logical(check.attributes), is.logical(ignore.col.order), is.logical(ignore.row.order), is.numeric(tolerance))
-  if (!is.data.table(target) || !is.data.table(current)) stop("'target' and 'current' must both be data.tables")
+  stopifnot(is.logical(trim.levels), is.logical(check.attributes), is.logical(ignore.col.order), is.logical(ignore.row.order), is.numeric(tolerance), is.data.table(target))
+  if (!is.data.table(current)) return(NextMethod())
```

Comment:
This will return `all.equal(data.table(a=1), data.frame(a=1))` to `TRUE` right? Shouldn't that be `FALSE` with reason "`current` is not a `data.table`"?

Reply:
It will return
```
[1] "Attributes: < Names: 2 string mismatches >"                              
[2] "Attributes: < Length mismatch: comparison on first 2 components >"       
[3] "Attributes: < Component 1: target is externalptr, current is character >"
[4] "Attributes: < Component 2: Modes: character, numeric >"                  
[5] "Attributes: < Component 2: Lengths: 2, 1 >"                              
[6] "Attributes: < Component 2: target is character, current is numeric >"    
```

which is the same (just with some terms swapped) as the result of `all.equal(data.frame(a=1), data.table(a=1))`